main.py

```diff
@@ -22,34 +39,104 @@
+    all: list[Any]
+
+
+class FxTweet(TypedDict):
+    """
+    Partial Tweet representation from Fx API
+    """
+
+    media: Optional[FxMedia]
+
+
+class FxResponse(TypedDict):
+    """
+    Top-level response from the Fx API
+    """
+
+    code: int
+    message: str
+    tweet: Optional[FxTweet]
+
+
+def get_fx_nagtype(user: str, tid: str) -> Optional[NagType]:
+    """
+    Request additional data from FxTwitter
+    to determine what embed type to use
+    """
+    try:
+        response = requests.get(
+            f"https://api.fxtwitter.com/{user}/status/{tid}", timeout=3
+        )
+        response.raise_for_status()
+        fx_res: FxResponse = response.json()
+        tweet = fx_res.get("tweet")
+        if tweet is None:
+            # tweet no longer exists
+            return None
+        media = tweet.get("media")
+        if media is None:
+            # text-only tweet
+            return None
+        media_all = media.get("all")
+        if len(media_all) == 1:
+            # original already embedded
+            return None
+        if len(media_all) == 2:
+            # special case for 2-image tweets
+            return NagType.SECOND_IMAGE
+        # every other case
+        return NagType.MOSAIC
+
+    except requests.HTTPError:
+        # can't access API, just be safe and respond with a mosaic
+        return NagType.MOSAIC
+
+
 @client.event
 async def on_ready() -> None:
     """Log client information on start"""
     print("Logged on as", client.user)
 
 
-async def nag(message: discord.Message) -> None:
+async def nag(message: discord.Message, nag_type: NagType) -> None:
     """
-    Post vxtwitter link in response to a native twitter link
+    Post normal fxtwitter link in response to a native twitter link
     """
-    tweets = twitter_url_regex.findall(message.content)
-    urls = [f"https://vxtwitter.com/{user}/status/{tid}" for user, tid in tweets]
+    tweets: list[tuple[str, str]] = twitter_url_regex.findall(message.content)
+    base = "https://fxtwitter.com"
+    modifier = ""
+    if nag_type is not NagType.FULL:
+        base = "https://g.fxtwitter.com"
+        if nag_type is NagType.SECOND_IMAGE:
+            modifier = "/photos/2"
+    urls = [f"{base}/{user}/status/{tid}{modifier}" for user, tid in tweets]
     if should_spoiler(message):
         urls = [f"|| {url} ||" for url in urls]
     if urls:
         nags[message.id] = await message.reply("\n".join(urls), mention_author=False)
-        if not should_nag(message):
+        if should_nag(message) is None:
             await unnag(message)
 
 
 async def unnag(message: discord.Message) -> None:
     """
-    Remove the vxtwitter link
+    Remove the fxtwitter link
     """
-    print(
-        f"!! removing response to {message.id} in {message.channel} on {message.guild}"
-    )
     if message.id in nags:
+        guild_string = ""
+        if message.guild is not None:
+            guild_string = f"on {message.guild.id}"
```

Comment:
This might be cleaner as a ternary, though their "improved" readability is somewhat debatable.
```py
guild_string = f"on {message.guild.id}" if message.guild is not None else ""
```

Also, it might be cleaner to test for boolean truthyness (simply `if message.guild`), as a `Guild` instance will never be falsy. On the other hand, this _is_ more explicit, and we know what The Zen of Python says about that.

Reply:
Done